File: StudentEnrollmentSystem/StudentEnrollmentSystem/GenerateData.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class GenerateData {
    public static void main(String[] args) {
        String[] firstNames = {"Tharindu", "Kasun", "Nimal", "Saman", "Kamal", "Ruwan", "Lahiru", "Nuwan", "Chathura", "Ishara"};
        String[] lastNames = {"Perera", "Silva", "Bandara", "Fernando", "de Silva", "Jayakody", "Gunawardena", "Weerasinghe", "Dissanayake", "Jayasinghe"};
        String[] courses = {"CS101", "SE202", "IT303", "IS404", "DS505", "ENG101", "MAT201"};
        String[] districts = {"Colombo", "Gampaha", "Kalutara", "Kandy", "Matale", "Galle", "Matara", "Kurunegala", "Anuradhapura", "Ratnapura"};
        
        Random rand = new Random();
        
        try (FileWriter writer = new FileWriter("enroll records.txt")) {
            writer.write("Student ID\tFull Name\tNIC Number\tCourse Code\tGPA\tDistrict\tIs Enrolled\tDate of Birth\n");
            
            for (int i = 1; i <= 150; i++) {
                String studentId = "STU" + (1000 + i);
                String fullName = firstNames[rand.nextInt(firstNames.length)] + " " + lastNames[rand.nextInt(lastNames.length)];
                String nic = (90 + rand.nextInt(10)) + "" + rand.nextInt(4) + "" + (100 + rand.nextInt(265)) + "" + (1000 + rand.nextInt(9000)) + "v";
                String course = courses[rand.nextInt(courses.length)];
                double gpa = 2.0 + (4.0 - 2.0) * rand.nextDouble();
                String district = districts[rand.nextInt(districts.length)];
                String isEnrolled = rand.nextBoolean() ? "TRUE" : "FALSE";
                String dob = (1 + rand.nextInt(12)) + "/" + (1 + rand.nextInt(28)) + "/199" + rand.nextInt(10);
                
                String record = String.format("%s\t%s\t%s\t%s\t%.2f\t%s\t%s\t%s\n", studentId, fullName, nic, course, gpa, district, isEnrolled, dob);
                writer.write(record);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: StudentEnrollmentSystem/StudentEnrollmentSystem/generate_data.py
import random

first_names = ["Tharindu", "Kasun", "Nimal", "Saman", "Kamal", "Ruwan", "Lahiru", "Nuwan", "Chathura", "Ishara",
               "Ashan", "Kavindu", "Sandun", "Ramesh", "Dinuka", "Sameera", "Nipun", "Harsha", "Janaka", "Sumudu",
               "Nadeeka", "Kumari", "Thilini", "Dinusha", "Nishanthi", "Kavindi", "Nethmi", "Oshadi", "Sewwandi", "Heshani"]

last_names = ["Perera", "Silva", "Bandara", "Fernando", "de Silva", "Jayakody", "Gunawardena", "Weerasinghe",
              "Dissanayake", "Jayasinghe", "Gamage", "Senanayake", "Herath", "Rathnayake", "Karunaratne",
              "Liyanage", "Fonseka", "Ekanayake", "Munasinghe", "Amarasinghe"]

courses = ["CS101", "SE202", "IT303", "IS404", "DS505", "ENG101", "MAT201"]
districts = ["Colombo", "Gampaha", "Kalutara", "Kandy", "Matale", "Nuwara Eliya", "Galle", "Matara", "Hambantota",
             "Jaffna", "Kilinochchi", "Mannar", "Vavuniya", "Mullaitivu", "Batticaloa", "Ampara", "Trincomalee",
             "Kurunegala", "Puttalam", "Anuradhapura", "Polonnaruwa", "Badulla", "Moneragala", "Ratnapura", "Kegalle"]

def generate_record(i):
    student_id = f"STU{1000 + i}"
    full_name = f"{random.choice(first_names)} {random.choice(last_names)}"
    nic_num = f"{random.randint(90, 99)}{random.randint(0, 3)}{random.randint(100, 365)}{random.randint(1000, 9999)}v"
    course = random.choice(courses)
    gpa = round(random.uniform(2.0, 4.0), 2)
    district = random.choice(districts)
    is_enrolled = random.choice(["TRUE", "FALSE"])
    dob = f"{random.randint(1, 12)}/{random.randint(1, 28)}/199{random.randint(0, 9)}"
    
    return f"{student_id}\t{full_name}\t{nic_num}\t{course}\t{gpa}\t{district}\t{is_enrolled}\t{dob}"

records = []
records.append("Student ID\tFull Name\tNIC Number\tCourse Code\tGPA\tDistrict\tIs Enrolled\tDate of Birth")
for i in range(1, 155):
    records.append(generate_record(i))

with open('c:/Users/user/OneDrive/Desktop/StudentEnrollmentSystem/enroll records.txt', 'w') as f:
    for record in records:
        f.write(record + "\n")
